gosdkexample/validator/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/gosdk"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/gosdk/command"
+)
+
+// This example implements a function that validate resources to ensure
+// spec.template.spec.securityContext.runAsNonRoot is set in workload APIs.
+
+func main() {
+	cmd := command.AsMain(context.TODO(), validator, command.Options{})
+	cmd.Short = "short description"
+	cmd.Long = "long description"
+	cmd.Example = "example"
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func validator(ctx context.Context, rl *gosdk.ResourceList) error {
+	return gosdk.ApplyFnBySelector(rl,
+		func(obj *gosdk.KubeObject) bool {
+			return obj.GetAPIVersion() == "apps/v1" && (obj.GetKind() == "Deployment" || obj.GetKind() == "StatefulSet" || obj.GetKind() == "DaemonSet" || obj.GetKind() == "ReplicaSet")
+		},
+		func(obj *gosdk.KubeObject) error {
+			var runAsNonRoot bool
+			found, err := obj.Get(&runAsNonRoot, "spec", "template", "spec", "securityContext", "runAsNonRoot")
+			if err != nil {
+				return gosdk.ErrorConfigObjectResult(err, obj)
```

Comment:
interesting that some errors are fmt errors and some are using gosdk.  Does it make sense to standardize error generation in all scenarios.

Reply:
it seems to me that the output is standardized to have results with messages.

Reply:
Updated them to make it more consistent.